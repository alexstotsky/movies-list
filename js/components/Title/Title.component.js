import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'

import { styleSheet } from '../../utils/commonPropTypes'


class Title extends React.PureComponent {
  static propTypes = {
    title: PropTypes.string.isRequired,
    containerStyle: styleSheet,
    titleStyle: styleSheet,
  }

  static defaultProps = {
    title: '',
  }

  render() {
    const { title, containerStyle, titleStyle } = this.props
    return (
      <View style={[styles.container, containerStyle]}>
        <Text style={[styles.title, titleStyle]}>
          {title}
        </Text>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    marginVertical: 5,
  },
  title: {
    fontSize: 22,
  },
})

export default Title
