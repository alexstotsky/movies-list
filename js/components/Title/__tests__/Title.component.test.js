import React from 'react'
import { shallow } from 'enzyme'

import Title from '../Title.component'

describe('Testing Subtitle component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Title title='' />,
    )
    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({ title: undefined })
    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({ title: 'asd' })
    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({ containerStyle: null })
    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({ containerStyle: 55 })
    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({ containerStyle: {} })
    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({ titleStyle: null })
    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({ titleStyle: 55 })
    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({ titleStyle: {} })
    expect(wrapper).toMatchSnapshot()
  })
})
