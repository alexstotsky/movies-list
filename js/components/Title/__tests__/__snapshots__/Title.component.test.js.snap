// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Subtitle component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 5,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 22,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Testing Subtitle component renders correctly 2`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 5,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 22,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Testing Subtitle component renders correctly 3`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 5,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 22,
        },
        undefined,
      ]
    }
  >
    asd
  </Text>
</View>
`;

exports[`Testing Subtitle component renders correctly 4`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 5,
      },
      null,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 22,
        },
        undefined,
      ]
    }
  >
    asd
  </Text>
</View>
`;

exports[`Testing Subtitle component renders correctly 5`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 5,
      },
      55,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 22,
        },
        undefined,
      ]
    }
  >
    asd
  </Text>
</View>
`;

exports[`Testing Subtitle component renders correctly 6`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 5,
      },
      Object {},
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 22,
        },
        undefined,
      ]
    }
  >
    asd
  </Text>
</View>
`;

exports[`Testing Subtitle component renders correctly 7`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 5,
      },
      Object {},
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 22,
        },
        null,
      ]
    }
  >
    asd
  </Text>
</View>
`;

exports[`Testing Subtitle component renders correctly 8`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 5,
      },
      Object {},
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 22,
        },
        55,
      ]
    }
  >
    asd
  </Text>
</View>
`;

exports[`Testing Subtitle component renders correctly 9`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 5,
      },
      Object {},
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 22,
        },
        Object {},
      ]
    }
  >
    asd
  </Text>
</View>
`;
